src/handlers/gestures/GestureDetector.tsx

```diff
@@ -79,20 +79,42 @@ function dropHandlers(preparedGesture: GestureConfigReference) {
   }
 }
 
+function checkGestureCallbacksForWorklets(gesture: GestureType) {
+  // if a gesture is explicitly marked to run on the JS thread there is no need to check
+  // if callbacks are worklets as the user is aware they will be ran on the JS thread
+  if (gesture.config.runOnJS) {
+    return;
+  }
+
+  const areAllWorklets = gesture.handlers.isWorklet.reduce(
+    (prev, current) => prev && (current === undefined || current === true),
+    true
+  );
+
+  const areAllNotWorklets = gesture.handlers.isWorklet.reduce(
+    (prev, current) => prev && (current === undefined || current === false),
+    true
+  );
+
+  if (!areAllWorklets && !areAllNotWorklets) {
```

Comment:
Maybe it would be simpler to just check:
```ts
gesture.handlers.isWorklet.includes(true) && gesture.handlers.isWorklet.includes(false)
```

Reply:
Done in https://github.com/software-mansion/react-native-gesture-handler/pull/1862/commits/2634a91aac5eb2db1b3b1f09b3eebbaa0a2d9eba.